src/systems/systems/src/electrical/battery.rs

```diff
@@ -0,0 +1,180 @@
+use super::{
+    ElectricalStateWriter, Potential, PotentialSource, PotentialTarget, ProvideCurrent,
+    ProvidePotential,
+};
+use crate::simulation::{SimulationElement, SimulatorWriter};
+use uom::si::{
+    electric_charge::ampere_hour, electric_current::ampere, electric_potential::volt, f64::*,
+};
+
+pub struct Battery {
+    writer: ElectricalStateWriter,
+    number: usize,
+    input: Potential,
+    charge: ElectricCharge,
+}
+impl Battery {
+    const MAX_ELECTRIC_CHARGE_AMPERE_HOURS: f64 = 23.0;
```

Comment:
I wonder if we'll want this to be slightly random in future, like e.g. the APU flap time. Might need some refactoring at that time to make this not a constant.

Reply:
I believe 23 _Ah_ is the maximum capacity for the batteries (though I'll definitely verify this in another PR which actually charges/discharges the battery). Of course as the battery ages the maximum somewhat reduces. We can store the battery age information in the EFB localstorage or something, and then load that age information in at the start of the simulation by adding another function to the `SimulationElement`. Something like `initialise(&mut self, reader: &mut SimulatorReader)`.